lectures/dynamic_programming/cake_eating_problem.md

```diff
@@ -0,0 +1,621 @@
+suitable discounting.
+
+If $c$ is chosen optimally using this trade off strategy, then we obtain maximal lifetime rewards from our current state $x$.
+
+Hence, $v(x)$ equals the right hand side of {eq}`bellman-cep`, as claimed.
+
+### An Analytical Solution
+
+It has been shown that, with $u$ as the CRRA utility function in
+{eq}`crra_utility`, the function
+
+```{math}
+:label: crra_vstar
+
+v^*(x_t) = \left( 1-\beta^{1/\gamma} \right)^{-\gamma}u(x_t)
+```
+
+solves the Bellman equation and hence is equal to the value function.
+
+You are asked to confirm that this is true in the exercises below.
+
+The solution {eq}`crra_vstar` depends heavily on the CRRA utility function.
+
+In fact, if we move away from CRRA utility, usually there is no analytical
+solution at all.
+
+In other words, beyond CRRA utility, we know that the value function still
+satisfies the Bellman equation, but we do not have a way of writing it
+explicitly, as a function of the state variable and the parameters.
+
+We will deal with that situation numerically when the time comes.
+
+Here is a Python representation of the value function:
+
+```{code-cell} julia
+function v_star(x, β, γ)
+
+    return (1 .- β .^ (1 ./ γ)) .^ (-γ) .* u(x, γ)
+end
+```
+
+And here's a figure showing the function for fixed parameters:
+
+```{code-cell} julia
+β, γ = 0.95, 1.2
+x_grid = LinRange(0.1, 5, 100) # note that we could also use range(), but directly constructing a LinRange should have less overhead (see https://docs.julialang.org/en/v1/base/collections/#Base.LinRange)
+
+plot(x_grid, v_star(x_grid, β, γ), label="value function", gridalpha = 0, xlabel = "x", xlabelfontsize = 12, legend_font = 12, legend_position =:topleft, linewidth = 1.5)
+```
+
+## The Optimal Policy
+
+Now that we have the value function, it is straightforward to calculate the
+optimal action at each state.
+
+We should choose consumption to maximize the
+right hand side of the Bellman equation {eq}`bellman-cep`.
+
+$$
+c^* = \arg \max_{c} \{u(c) + \beta v(x - c)\}
+$$
+
+We can think of this optimal choice as a function of the state $x$, in
+which case we call it the **optimal policy**.
+
+We denote the optimal policy by $\sigma^*$, so that
+
+$$
+\sigma^*(x) := \arg \max_{c} \{u(c) + \beta v(x - c)\}
+\quad \text{for all } x
+$$
+
+If we plug the analytical expression {eq}`crra_vstar` for the value function
+into the right hand side and compute the optimum, we find that
+
+```{math}
+:label: crra_opt_pol
+
+\sigma^*(x) = \left( 1-\beta^{1/\gamma} \right) x
+```
+
+Now let's recall our intuition on the impact of parameters.
+
+We guessed that the consumption rate would be decreasing in both parameters.
+
+This is in fact the case, as can be seen from {eq}`crra_opt_pol`.
+
+Here's some plots that illustrate.
+
+```{code-cell} julia
+function c_star(x, β, γ)
+
+    return (1 .- β .^ (1 ./ γ)) .* x
+end
+```
+
+Continuing with the values for $\beta$ and $\gamma$ used above, the
+plot is
+
+```{code-cell} julia

```

Comment:
For my tastes, I think moving the `x_grid` down to this point is probably better.  This is the place where you can use the broadcasting.  i.e. `c_star.(x_grid, beta, gamma)` etc.

Reply:
Thanks, the comments are very helpful.